lib/_http_outgoing.js

```diff
@@ -79,7 +79,7 @@ exports.OutgoingMessage = OutgoingMessage;
 
 
 OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
-  if (callback)
+  if (typeof callback === 'function')
```

Comment:
Maybe it should be a TypeError when `typeof callback !== 'function'`?  This change silently ignores non-functions, which seems like a good way for bugs in user code to go unnoticed.